packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2055,15 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (this.match(tt._with) && !this.hasPlugin("moduleAttributes")) {
+      this.expectPlugin("moduleAttributes");
+    } else if (this.eat(tt._with)) {
+      return this.parseKeyValuePairs();
+    }
+    return [];
```

Comment:
I think that we should do something like this:
1. If there is `with`,
  i. Expect the plugin
  ii. return `parseKeyValuePairs`
2. If there is the plugin
  i. Return `[]`

By doing so, we don't return an empty array if the plugin is disabled (since disabled plugins shouldn't have any effect on the AST).
Even better, we should only set `node.attributes` if `maybeParseModuleAttributes` returns the array.

Reply:
Understood. That makes sense. I will make the changes accordingly.

Reply:
updated the code with these changes.